packages/babel-plugin-htm/index.mjs

```diff
@@ -85,48 +77,49 @@ export default function htmBabelPlugin({ types: t }, options = {}) {
 			child = child.trim();
 		}
 		if (typeof child==='string') {
-			const matches = child.match(/\$\$\$_h_\[(\d+)\]/);
-			if (matches) return currentExpressions[matches[1]];
 			return stringValue(child);
 		}
 		return child;
 	}
 
 	function h(tag, props) {
 		if (typeof tag==='string') {
-			const matches = tag.match(/\$\$\$_h_\[(\d+)\]/);
-			if (matches) tag = currentExpressions[matches[1]];
-			else tag = t.stringLiteral(tag);
+			tag = t.stringLiteral(tag);
 		}
 
-		//const propsNode = props==null || Object.keys(props).length===0 ? t.nullLiteral() : t.objectExpression(
-		const propsNode = t.objectExpression(
-			Object.keys(props).map(key => {
-				let value = props[key];
-				if (typeof value==='string') {
-					const tokenizer = /\$\$\$_h_\[(\d+)\]/g;
-					let token, lhs, root, index=0, lastIndex=0;
-					const append = expr => {
-						if (lhs) expr = t.binaryExpression('+', lhs, expr);
-						root = lhs = expr;
-					};
-					while ((token = tokenizer.exec(value))) {
-						append(t.stringLiteral(value.substring(index, token.index)));
-						append(currentExpressions[token[1]]);
-						index = token.index;
-						lastIndex = tokenizer.lastIndex;
-					}
-					if (lastIndex < value.length) {
-						append(t.stringLiteral(value.substring(lastIndex)));
+		let propsNode;
+
+		if (t.isObjectExpression(props)) {
+			propsNode = props;
+			for (let i in props) {
+				if (props.hasOwnProperty(i) && props[i] && props[i].type) {
+					for (let j=0; j<props.properties.length; j++) {
+						if (props.properties[j].start > props[i].start) {
+							props.properties.splice(j, 0, t.objectProperty(propertyName(i), props[i]));
+							break;
+						}
 					}
-					value = root;
-				}
-				else if (typeof value==='boolean') {
-					value = t.booleanLiteral(value);
+					delete props[i];
 				}
-				return t.objectProperty(propertyName(key), value);
-			})
-		);
+			}
+		}
+		else {
+			propsNode = t.objectExpression(
+				Object.keys(props).map(key => {
+					let value = props[key];
+					if (typeof value==='string') {
+						value = t.stringLiteral(value);
+					}
+					else if (typeof value==='boolean') {
+						value = t.booleanLiteral(value);
+					}
+					else if (typeof value==='number') {
+						value = t.stringLiteral(value + '');
+					}
```

Comment:
Is the `typeof value==='number'` case expected to happen, or could it be removed?